vendor/README.md

```diff
@@ -0,0 +1,353 @@
+Vendor tests
+============
+
+When creating a badge for a new service or changing a badge's behavior, tests
+should be included. They serve three purposes:
+
+1. The contributor and reviewer can easily verify the code works as
+   intended.
+
+2. When a badge stops working on the live server, maintainers can find out
+   right away.
+
+3. They speed up future contributors when they are debugging or improving a
+   badge.
+
+Contributors should take care to cover each part of a badge's functionality,
+and ideally, all code branches:
+
+1. Typical case
+  - File is present
+  - Build fails/succeeds
+2. Expected resource not found
+  - Vendor may provide 200 error code with different response format
+  - Vendor may return a 404 or other >= 400 status code
+3. Customization
+  - Non-default parameters like tags and branches
+4. Server errors and other malformed responses
+  - Vendor may return status code 500 and higher
+  - Invalid JSON
+  - Attributes missing or have incorrect types
+  - Headers missing
+5. Connection errors
+
+Tutorial
+--------
+
+In this tutorial, we'll write tests for the Travis badge.
+Here, you can see the [source code][travis-example]:
+
+```js
+camp.route(/^\/travis(-ci)?\/([^\/]+\/[^\/]+)(?:\/(.+))?\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var userRepo = match[2];  // eg, espadrine/sc
+  var branch = match[3];
+  var format = match[4];
+  var options = {
+    method: 'HEAD',
+    uri: 'https://api.travis-ci.org/' + userRepo + '.svg',
+  };
+  if (branch != null) {
+    options.uri += '?branch=' + branch;                         // (3)
+  }
+  var badgeData = getBadgeData('build', data);
+  request(options, function(err, res) {
+    if (err != null) {
+      console.error('Travis error: ' + err.stack);              // 5
+      if (res) { console.error(''+res); }
+      badgeData.text[1] = 'invalid';
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      var state = res.headers['content-disposition']
+                     .match(/filename="(.+)\.svg"/)[1];
+      badgeData.text[1] = state;
+      if (state === 'passing') {
+        badgeData.colorscheme = 'brightgreen';                  // 1
+      } else if (state === 'failing') {
+        badgeData.colorscheme = 'red';                          // 1
+      } else {
+        badgeData.text[1] = state;                              // 1, 2, 3
+      }
+      sendBadge(format, badgeData);
+
+    } catch(e) {
+      badgeData.text[1] = 'invalid';                            // 4
+      sendBadge(format, badgeData);
+    }
+  });
+}));
+```
+
+It handles the typical cases (1), resource not found (2), customization (3),
+malformed responses (4), and connection errors (5).
+
+Before getting started, install the project dependencies if you haven't
+already:
+
+```
+npm i
```

Comment:
It's explained right before. If they are still confused it's an easy answer to find in Google.. In scripts I use long form of commands and options for readability's sake, but this I'm expecting to be typed.